conf/imx_hab4/README.txt

```diff
@@ -0,0 +1,268 @@
+by U-boot: on the imx7ulpea this address should be 0x2f010000.
+
+This information will be used to generate the Command Sequence File
+(CSF) used to sign the image.
+
+mod_4_mfgtool.sh
+-----------------
+#!/bin/bash
+if [ $# -lt 2 ] || [ ! -e $2 ]; then
+        echo You must provide an action and a valid u-boot file as parameters
+        echo Example: $0 clear_dcd_addr u-boot.imx
+        exit 1
+fi
+
+# DCD address must be cleared for signature, as mfgtool will clear it.
+if [ "$1" == "clear_dcd_addr" ]; then
+        # store the DCD address
+        dd if=$2 of=dcd_addr.bin bs=1 count=4 skip=12
+        # generate a NULL address for the DCD
+        dd if=/dev/zero of=zero.bin bs=1 count=4
+        # replace the DCD address with the NULL address
+        dd if=zero.bin of=$2 seek=12 bs=1 conv=notrunc
+        rm zero.bin
+fi
+
+# DCD address must be set for mfgtool to localize the DCD table.
+if [ "$1" == "set_dcd_addr" ]; then
+        # restore the DCD address with the original address
+        dd if=dcd_addr.bin of=$2 seek=12 bs=1 conv=notrunc
+        rm dcd_addr.bin
+fi
+
+The Command Sequence File will need access to the SRK table amd PEM files.
+You can get them as before from the lmp-manifest repo.
+
+SPLMfg.csf
+----------
+[Header]
+Version = 4.1
+Security Configuration = Open
+Hash Algorithm = sha256
+Engine Configuration = 0
+Certificate Format = X509
+Signature Format = CMS
+Engine = CAAM
+
+[Install SRK]
+File = "./SRK_1_2_3_4_table.bin"
+Source index = 0
+
+[Install CSFK]
+File = "./CSF_1_crt.pem"
+
+[Authenticate CSF]
+
+[Install Key]
+# Key slot index used to authenticate the key to be installed
+Verification index = 0
+
+# Key to install
+Target index = 2
+File = "./IMG_1_crt.pem"
+
+[Authenticate Data]
+Verification index = 2
+Blocks = 0x2f010000 0x02c 0x00258 "./SPLmfg.bin"
+
+[Authenticate Data]
+Verification index = 2
+Blocks = 0x2f010400 0x000 0x21c00 "./SPLmfg.bin"
+
+To sign the SPLMfg image just copy the SPL image from U-boot, rename it to
+SPLMfg.bin and execute the following script (with the two previous ones in the 
+path as well as the necessary tables and keys).
+
+Also notice that you will also need to download the CST tool from IMX
+(in the script below, we used the cst_64 bit version).
+
+sign.spl.mfg.sh
+----------------
+#!/bin/bash
+PROG_NAME=SPLmfg
+
+# DCD address must be cleared for signature, as mfgtool will clear it.
+./mod_4_mfgtool.sh clear_dcd_addr SPLmfg.bin
+
+# generate the signatures, certificates, ... in the CSF binary
+./cst_64 --o SPLmfg_csf.bin --i SPLmfg.csf
+
+# DCD address must be set for mfgtool to localize the DCD table.
+./mod_4_mfgtool.sh set_dcd_addr SPLmfg.bin
+
+# gather ${PROG_NAME} + its CSF
+cat SPLmfg.bin SPLmfg_csf.bin > SPLmfg_signed.bin
+
+==============================
+Boot a Signed SPLmfg using SDP
+==============================
+Use a version of the UUU tool that includes support for the -dcdaddr
+and -cleardcd flags; these  were merged in November 2019. Then simply
```

Comment:
Probably good to point the link to the git changes instead of just saying that they were merged in november.